integration-test-rex/src/test/java/org/jboss/pnc/integrationrex/RemoteServices.java

```diff
@@ -0,0 +1,161 @@
+import org.jboss.util.StringPropertyReplacer;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import static org.jboss.pnc.common.json.moduleconfig.microprofile.SchedulerMicroprofileConfig.SCHEDULER_URL_KEY;
+
+public class RemoteServices {
+
+    private static final Logger logger = LoggerFactory.getLogger(RemoteServices.class);
+
+    @Rule
+    public BPMWireMock bpm = new BPMWireMock(8088);
+
+    protected static String authServerUrl;
+
+    protected static String keycloakRealm = "newcastle-testcontainer";
+
+    private static dasniko.testcontainers.keycloak.KeycloakContainer keycloakContainer;
+    private static GenericContainer rexContainer;
+
+    @BeforeClass
+    public static void beforeAll() throws IOException {
+        logger.info("Starting containers ...");
+        startRemoteServices();
+
+        // 8080 IS JBOSS CONTAINER
+        // 8088 IS BPM WIREMOCK MOCK
+        Testcontainers.exposeHostPorts(8080, 8088);
+        logger.info("Containers started.");
+    }
+
+    @Deployment(testable = false)
+    public static EnterpriseArchive deploy() throws IOException {
+        return Deployments.testEar();
+    }
+
+    @AfterClass
+    public static void afterAll() throws InterruptedException {
+        logger.info("Stopping containers ...");
+        keycloakContainer.stop();
+        rexContainer.stop();
+        logger.info("Containers stopped.");
+        Thread.sleep(1000L); // make sure all resources are released
+    }
+
+    protected static void startRemoteServices() throws IOException {
+        // properties to share port numbers
+        Properties testProperties = new Properties();
+        try (InputStream propFile = new FileInputStream("target/test.properties")) {
+            testProperties.load(propFile);
+        }
+
+        Network network = Network.newNetwork();
+
+        Consumer<OutputFrame> keycloakLogConsumer = frame -> logger.debug("KEYCOAK >>" + frame.getUtf8String());
+
+        String keycloakHostPort = testProperties.getProperty(GetFreePort.KEYCLOAK_PORT);
+        String keycloakPortBinding = keycloakHostPort + ":" + 8080; // 8080 is in-container port
+        keycloakContainer = new CustomKeycloakContainer("quay.io/keycloak/keycloak:21.1.0").withNetwork(network)
```

Comment:
Refactor to seperate file?

Reply:
I think it's fine, no need to do it yet.